### PR TITLE
#161966897 Test Login via Social Authentication

### PR DESCRIPTION
**Pull request by codeplus254**
#### What does this PR do?
Tests the login via Google, Twitter or Facebook
#### Description of Task to be completed?
Write failing tests before the login feature is implemented. 
This will be done by updating the serializers.py, urls.py and views.py
#### How should this be manually tested?

- After cloning the repo, CD into it. 

- Create and start your virtual environment 

- Install dependencies. `pip3 install -r requirements.txt`

- Define the following environment variables and place them in a `.env` file in the root directory

`export DATABASE_USER='postgres'
export DATABASE_PASSWORD='postgres'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='test_db' `

- Add the variables into the virtual environment you just created. `source .env`

- run `python3 manage.py test`. If the tests cannot be found automatically, you'll need to add the url after the word test (  `python3 manage.py test authors.apps.authentication.tests.social_authentication_tests` )
#### Any background context you want to provide?
The user will have options for logging in directly into the app through Facebook, Google or Twitter
#### What are the relevant pivotal tracker stories?
[#161966897](https://www.pivotaltracker.com/story/show/161966897)
#### Screenshots (if appropriate)
screenshot 2018-12-07 at 09 26 23
screenshot 2018-12-07 at 09 27 18

### CODE REVIEW

**Comment by ThaDeveloper** [comment]
Good thing there are zero errors in the tests. Include how to install dependencies and run migrations.

**Comment by r-wambui** [comment]
@codeplus254 Please rebase develop

File: .gitignore [hunk 1]
```diff
@@ -92,3 +94,6 @@ ENV/
 
 # SQLite3
 db.sqlite3
+
+#Django Migrations
+0001_initial.py
```
Comment (r-wambui) [inline .gitignore]: Why did you add migrations to gitignore?

Comment (codeplus254) [reply .gitignore]: @r-wambui the PR couldn't be merged automatically with that file in...plus it's autogenerated by django

Comment (r-wambui) [reply .gitignore]: But we do not ignore migrations. You could have looked for a way to solve the conflicts.

File: authors/apps/authentication/tests/social_authentication_tests.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+"""This module runs tests for user login process via Facebook, Twitter or Google"""
+import json
```
Comment (r-wambui) [inline authors/apps/authentication/tests/social_authentication_tests.py]: json is imported but not used

File: authors/apps/authentication/tests/social_authentication_tests.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        data = {
+            "provider": "twitter", 
+            "access_key": self.twitter_key, 
+            "access_key_secret": self.twitter_secret}
+        response = self.client.post('api/socialAuth', data=data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_facebook_login_no_tokens(self):
+        """this method test that a user cannot login via Facebook if there's no access_key and access_key_secret"""
```
Comment (r-wambui) [inline authors/apps/authentication/tests/social_authentication_tests.py]: Good use of docstrings. However you did not follow pep8. 
https://www.python.org/dev/peps/pep-0008/#maximum-line-length

File: .gitignore (hunk 1 above)
Comment (codeplus254) [reply .gitignore]: Ok let me find a way around this. :)

File: authors/apps/authentication/tests/social_authentication_tests.py (hunk 3 above)
Comment (codeplus254) [reply authors/apps/authentication/tests/social_authentication_tests.py]: Sorry, I had Pylinted the file but somewhere in the rebase or squash or sth I lost the changes.

File: authors/apps/authentication/tests/social_authentication_tests.py (hunk 2 above)
Comment (codeplus254) [reply authors/apps/authentication/tests/social_authentication_tests.py]: I don't know what happened. Pylint had noted this and I changed but the changes were lost in some git process I ran

File: authors/settings.py [hunk 4]
```diff
@@ -43,7 +43,16 @@
     'authors.apps.authentication',
     'authors.apps.core',
     'authors.apps.profiles',
+
+    'social_django'
```
Comment (r-wambui) [inline authors/settings.py:47]: Should we be having social-django in the requirements.txt as well?

Comment (codeplus254) [reply authors/settings.py:47]: Yes I will add it.

**Comment by rkemmy** [comment]
@codeplus254 you can add an  `__init__.py` file to your test folder so that it is considered a package and you can then run tests using `python manage.py test`
